presto-main/src/test/java/com/facebook/presto/cost/TestSimpleFilterProjectSemiJoinStatsRule.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.facebook.presto.sql.planner.plan.Assignments;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.expression;
+
+public class TestSimpleFilterProjectSemiJoinStatsRule
+{
+    private StatsCalculatorTester tester;
+
+    private SymbolStatsEstimate aStats = SymbolStatsEstimate.builder()
+            .setLowValue(0)
+            .setHighValue(10)
+            .setDistinctValuesCount(10)
+            .setNullsFraction(0.1)
+            .build();
+
+    private SymbolStatsEstimate bStats = SymbolStatsEstimate.builder()
+            .setLowValue(0)
+            .setHighValue(100)
+            .setDistinctValuesCount(10)
+            .setNullsFraction(0)
+            .build();
+
+    private SymbolStatsEstimate cStats = SymbolStatsEstimate.builder()
+            .setLowValue(5)
+            .setHighValue(30)
+            .setDistinctValuesCount(2)
+            .setNullsFraction(0.5)
+            .build();
+
+    private SymbolStatsEstimate expectedFilteredPositiveAStats = SymbolStatsEstimate.builder()
```

Comment:
field name could expose the fact this is for a symbol when doing `a in (.. c ..)`